### DISCUSSION TITLE
[Idea] Implement mobile version of tables

### DISCUSSION THREAD

**Discussion by Hlavtox** [PrestaShop/PrestaShop#32280] [Ideas]
#### Is your feature request related to a problem?
Tables are unsuable on mobile devices. Alternative layout should be implemented.

#### Solution 
- We should implement alternative layout version to all tables.
- This layout could be triggered by adding something like table-mobile-sm, table-mobile-md, table-mobile-lg or table-mobile-xl. When going under the breakpoint, table will change to mobile view. This way it's going to be universal. Order page grid will go to mobile ASAP, some narrower grids can go later.
- Setting `display: block;` on `td` and `th` of the table will stack the cells to one column.
- We need to add some labels to filters and cells. This can be done by rendering them hidden to every cell (big code) or render them once to header and show them by CSS ::before (PS 1.6 solution)
- Then it's only a matter of putting some simple styling on it - separating the rows by some nice color or border, adjusting spacing and hiding column labels on the top.

#### Screenshot
![localhost_prestashop_admin-dev_index php_sell_orders___token=x5iZEIqW6fVaRDkrQWjNhoOwTb-3NbOgjFYrUl1UQbU](https://user-images.githubusercontent.com/6097524/125178315-fb027480-e1e3-11eb-9d7a-5e53534bdec0.png)

![1 6](https://user-images.githubusercontent.com/6097524/125178318-fccc3800-e1e3-11eb-942c-8bbb8e271652.JPG)

**Comment by hibatallahAouadni** [comment]
Hello @Hlavtox 

Thank you for your suggestion. The Product Team will take it into consideration for future developments. 
Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you create it. 

If you are willing to implement this feature into PrestaShop as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/), the teams will be happy to help you by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know! 

Thank you

**Comment by matks** [comment]
ping @NeOMakinG I think it's rather a developer topic

**Comment by NeOMakinG** [comment]
There are some mockups by @prestascott about this, it's To Do thing obviously!

**Comment by matks** [comment]
> There are some mockups by @prestascott about this, it's To Do thing obviously!

If @prestascott is OK, I think we can remove the `Waiting for PM` for this and add it as `To Do` 😉

**Comment by prestascott** [comment]
Hi everyone,

Thank you @Hlavtox for your mockups. 
Unfortunately this style of table responsive on mobile isn't adapted. As you can see, search inputs are taking way too much space.  

Instead of having this type of table
![table1](https://user-images.githubusercontent.com/54888056/131364891-2bafc9b0-ef06-492e-b77c-1fdbd70a6e5d.gif)

I suggest to have this one and make the checkbox and Id columns sticky for example
![table2](https://user-images.githubusercontent.com/54888056/131364949-ac4913f1-3a76-4b90-a7d6-8ae1e6a6f2da.gif)

Or we can keep your mockups but change the filters and make it simplier :)

Wdyt?

**Comment by Hlavtox** [comment]
@prestascott I would definitely keep the first solution, exactly as you designed it in front office. ;-)

![Výstřižek](https://user-images.githubusercontent.com/6097524/136161347-1611a97a-23b5-4f64-8e32-5a79e9aed20e.JPG)
_____________

The second solution would be bad, you don't know what you are searching for in the table - should name be sticky, id, reference? Sometimes the older proven solution is the best. :-)

**Comment by Hlavtox** [comment]
@prestascott I'm gonna do it like this for now, we can improve on the fly. Cell labels could be added also.

![mobile table](https://user-images.githubusercontent.com/6097524/142387507-1f5351df-3f7d-43d8-a1d9-b4628526544b.JPG)

**Comment by prestascott** [comment]
What happens when you click on the "Sort" button and the "Filter" button? @Hlavtox

**Comment by Hlavtox** [comment]
@prestascott It should open a collapse with filters. Or we can just keep them displayed, but I wanted the users to see the data first.

**Comment by prestascott** [comment]
Can you provide us screenshots so we can see please?

**Comment by prestascott** [comment]
Here is my proposition:
- I added the search field so users can still type what they want they are looking for
- "Sort by" and "Filter by" buttons are smaller so we can save some space
- When users click the "Filter by" button, it opens a page where they can manage filters
- Labels have been added because it's more clear, only the text isn't really understandable (cf. "No" for "New client"). It needs context.

<img width="375" alt="resize-01" src="https://user-images.githubusercontent.com/54888056/142440857-01ef8c06-dc97-49f0-93cd-63c817d16ab8.png">
<img width="375" alt="resize-02" src="https://user-images.githubusercontent.com/54888056/142440864-71ef8b0e-de5a-4599-8069-0a21eeec2e2b.png">
<img width="375" alt="resize-03" src="https://user-images.githubusercontent.com/54888056/142440868-3a565f09-654f-452b-9e37-08874ed3684c.png">
<img width="375" alt="resize-04" src="https://user-images.githubusercontent.com/54888056/142440873-724ea99e-7aed-4563-b71b-7981e279cb83.png">
<img width="375" alt="resize-05" src="https://user-images.githubusercontent.com/54888056/142440881-e443f63d-1cde-4cb8-9d82-b43ce7e4f6b8.png">

Wdyt?

**Comment by Hlavtox** [comment]
@prestascott Very nice, lets go! 

Will need some help from @NeOMakinG though 🙏😎

**Comment by prestascott** [comment]
Okay, there are some mockups to finalize on our side such as the Sort by behaviors. 
I will upload them in the comments next week!

**Comment by NeOMakinG** [comment]
I'm up to help you @Hlavtox but can we try to finish the availability component before? Let's speak about it on slack!

**Comment by prestascott** [comment]
Hello everyone, I'm back with the latest mockups. 
You can also access to the [Figma file](https://www.figma.com/file/aCBf2HlypjT0ZxP2XcsT68/Core-Github-issues?node-id=1041%3A8876).

**DEFAULT**
<img width="375" alt="default (1)" src="https://user-images.githubusercontent.com/54888056/143261093-098f4151-718a-447b-9ed2-d2a6ad198c85.png">

**FILTER BY**
Behaviors: 
- Opens a page placed on top 
- Once a filter is selected, the "Apply" buttons is enable
<img width="1192" alt="Capture d’écran 2021-11-24 à 15 55 59" src="https://user-images.githubusercontent.com/54888056/143261384-92cfa213-86a4-43e8-a054-415a02e66739.png">

**FILTERS APPLIED AND BULK ACTIONS**
Behaviors:
- The number of filters applied appears on the "Filter by" button
- The filters applied appears as tags that can be removed by clicking on the cross icon
- The "Clear filters" text button allows users to remove all filters applied
- When an order is checked, the order is immediately highlighted by purple outlines, and the "Bulk actions" button is enable
<img width="462" alt="Capture d’écran 2021-11-24 à 15 56 34" src="https://user-images.githubusercontent.com/54888056/143261502-49e14cc8-8876-424b-bc69-09b768580cdf.png">

**SORT BY**
Behaviors: 
- Like the "Filter by" feature, it opens a page placed on top
- No sorting is applied by default
- Sorting by ID can be Ascending or Descending
- Sorting by Name can be A-Z or Z-A
<img width="950" alt="Capture d’écran 2021-11-24 à 16 00 14" src="https://user-images.githubusercontent.com/54888056/143262062-b6305dc0-7369-41b5-90cf-a312a239cd3c.png">

**SORTING IS APPLIED**
Behaviors:
- The "Sort by" button is highlighted with purple outlines
- The sorting applied appears as a tag that can be removed by clicking on the cross icon 
<img width="297" alt="Capture d’écran 2021-11-24 à 16 03 15" src="https://user-images.githubusercontent.com/54888056/143262589-63147ee8-4f65-4aed-81d8-ef069cc403cc.png">

FYI: I discussed with @Julievrz mainly about wording. We decided to change the "Reset" label by "Clear filters" because it's more explicit.

Let me know wdyt :)

**Comment by Hlavtox** [comment]
@prestascott Scott, it looks absolutely awesome!!!

One small thing: We are not able to do the search row for now. There is no such feature now. It will be part of the filter modal.

![143261384-92cfa213-86a4-43e8-a054-415a02e66739](https://user-images.githubusercontent.com/6097524/144474052-69d39450-7a04-484e-b7bf-bbc852313573.png)

**Comment by prestascott** [comment]
@Hlavtox Thanks a lot!

Oh no... Then it will be difficult to provide a good UX if the search is hidden inside the filter. Search, sort, and filters are 3 different things. 

First, users search by typing.
Then, users filter or sort the results.

It's very weird and unusual to do a search in the filter.

**Comment by Hlavtox** [comment]
@prestascott Scott, there is no such "search" feature on desktop. It's only filters - text input, range or select. And merchants are used to how it works on desktop. 

Also, with only one search input, you would limit their ability to filter. On desktop, you can filter by multiple text fields. So you can for example search for orders by customer name and payment methods (both are text inputs on desktop), with your design, you can't do that.

**Comment by Hlavtox** [comment]
> It's very weird and unusual to do a search in the filter.

Man, how can you say that it's weird if people are using it the same way for at least 10 years? 😄 Don't create problems that do not exist, please. Or like have merchants IQ 40 or are merchants monkeys? I really don't understand.

Prestashop 1.4, release date 2011
![1-zoom](https://user-images.githubusercontent.com/6097524/144627413-3ae1d8a3-cb2a-41a6-b2d3-0133c3dd0d4b.jpg)

**Comment by hibatallahAouadni** [comment]
Hello @prestascott 

Let me summarize what @Hlavtox want to say :sweat_smile: 
So, if I want make a search by **`Customer`** or **`Total`** or **`Payment`**:

![image](https://user-images.githubusercontent.com/23308427/144840577-68fab08c-1204-4a76-acbf-a922d27d0dae.png)

I should click on this Select, click on one of this attribute, then type the value:

![image](https://user-images.githubusercontent.com/23308427/144840327-b4f72584-8220-436c-9239-ec1f344cc243.png)

But, what if I want to make search by **`Customer`** name and the **`Total`** of the order or **`Customer`** name and the **`Payment`** method? :thinking: 
With the provided design, it's impossible!!
Please check and feedback.

Thanks!

**Comment by ttoine** [comment]
Dear @Hlavtox , we appreciate your active participation in the project, and have enjoyed collaborating with you so far. However, your last [answer](https://github.com/PrestaShop/PrestaShop/issues/25313#issuecomment-985607309) is rude and disrespectful of the [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md). You may not agree with a solution that we proposed but this type of message will not help us get things done. On the opposite: after reading it, we absolutely do not want to answer you and by the way, we have no obligation to do so. 

You are right, there is a problem with the search design and there is no way to search in different columns with this design. But if you want an answer and if you want us to continue working with you, please stop being disrespectful and keep the conversation polite and professional. Otherwise, we will be forced to stop working with you.

As you may know, the role of Product Managers and Product Designers is to find solutions that meet everyone's needs. We know that being the only ones with the decision-making power on the acceptance or rejection of contributions bringing about a functional change is problematic because the waiting times are sometimes long. We are going to discuss this subject internally in the next few days to find solutions.

We hope to be able to continue working with you in a respectful way.

**Comment by Hlavtox** [comment]
@ttoine Can you please specify which part of Code of Conduct did I break?

I only said that merchants are **not stupid** and we shouldn't looked at them as they were unable to use a table filter they use for 10 years! If something, I stood up for them!

So please, stop being so touchy and affected by everything I say, I don't have time or mood for discussing this for a millionth time. If you have a problem with me, I will close all my issues and PRs and go. 👍

[a comment by an automated account is not shown]

**Comment by ttoine** [comment]
The problem @Hlavtox is that you are playing with the limits of the Code of Conduct.

So, to be clear, if you want to have your improvements merged into the project:
- **be friendly**, it helps to create good relationship and foster the review (e.g: cool, I like this guy, I will review their pull request)
- **write good code**, respect the best practices and the guidelines of the project (keep in mind that the project must be consistent and maintainable)
- **be patient**, because there are a loooot of contributions (it takes time to check and test them to ensure it is useful and it doesn't break anything and everybody is in best effort)
- **be friendly**, polite, respectful, tolerant (again and always)

**Comment by ttoine** [comment]
In one quick sentence: everybody here need you to be friendly.
